rust/src/shards/gui/layouts/table.rs

```diff
@@ -0,0 +1,429 @@
+  }
+
+  fn help(&mut self) -> OptionalString {
+    OptionalString(shccstr!("Table layout."))
+  }
+
+  fn inputTypes(&mut self) -> &Types {
+    &ANYS_TYPES
+  }
+
+  fn outputTypes(&mut self) -> &Types {
+    &ANYS_TYPES
+  }
+
+  fn parameters(&mut self) -> Option<&Parameters> {
+    Some(&TABLE_PARAMETERS)
+  }
+
+  fn setParam(&mut self, index: i32, value: &Var) -> Result<(), &str> {
+    match index {
+      0 => {
+        let seq = Seq::try_from(value)?;
+        for shard in seq.iter() {
+          let mut s = ShardsVar::default();
+          s.set_param(&shard).unwrap();
+          self.shards.push(s);
+        }
+        Ok(self.rows.set_param(value))
+      }
+      1 => {
+        let columns = Seq::try_from(value)?;
+        for column in columns.iter() {
+          let column: crate::types::Table = column.as_ref().try_into()?;
+          if let Some(header) = column.get_static(cstr!("Header")) {
+            match header.valueType {
+              SHType_String => {
+                self.header_shards.push(None);
+              }
+              SHType_ShardRef | SHType_Seq => {
+                let mut s = ShardsVar::default();
+                s.set_param(&header).unwrap();
+                self.header_shards.push(Some(s));
+              }
+              _ => unreachable!(),
+            }
+          } else {
+            self.header_shards.push(None);
+          }
+        }
+        Ok(self.columns.set_param(value))
+      }
+      2 => Ok(self.striped.set_param(value)),
+      3 => Ok(self.resizable.set_param(value)),
+      4 => Ok(self.row_index.set_param(value)),
+      _ => Err("Invalid parameter index"),
+    }
+  }
+
+  fn getParam(&mut self, index: i32) -> Var {
+    match index {
+      0 => self.rows.get_param(),
+      1 => self.columns.get_param(),
+      2 => self.striped.get_param(),
+      3 => self.resizable.get_param(),
+      4 => self.row_index.get_param(),
+      _ => Var::default(),
+    }
+  }
+
+  fn requiredVariables(&mut self) -> Option<&ExposedTypes> {
+    self.requiring.clear();
+
+    // Add UI.Parents to the list of required variables
+    util::require_parents(&mut self.requiring, &self.parents);
+
+    Some(&self.requiring)
+  }
+
+  fn exposedVariables(&mut self) -> Option<&ExposedTypes> {
+    self.exposing.clear();
+
+    let mut exposed = false;
+    for s in &self.shards {
+      exposed |= util::expose_contents_variables(&mut self.exposing, s);
+    }
+
+    if exposed {
+      Some(&self.exposing)
+    } else {
+      None
+    }
+  }
+
+  fn hasCompose() -> bool {
+    true
+  }
+
+  fn compose(&mut self, data: &InstanceData) -> Result<Type, &str> {
+    // compose header shards
+    for header_shard in &mut self.header_shards {
+      if let Some(header_shard) = header_shard {
+        header_shard.compose(data)?;
+      }
+    }
+
+    // compose row shards
+    let mut data = *data;
+    let mut shared: ExposedTypes;
+
+    // we need to inject the row index to the inner shards
```

Comment:
if you really really really need a variable you could inject one like I think `Map` or `ForEach` does which is a special character, to check in `core.cpp`

Reply:
I don't see anything there that has the same functionality.